### ISSUE TITLE
`native.apple_static_library` doesn't propagate execution tags (e.g. `no-remote`) with `--experimental_allow_tags_propagation`

### ISSUE THREAD

**Issue by brentleyjones** [bazelbuild/bazel#14214]
### Description of the problem / feature request:

Most of the native apple rules, such as `native.apple_static_library` and the native linking, don't propagate execution tags (e.g. `no-remote`) when using `--experimental_allow_tags_propagation`. This makes it hard to modify their execution info, since `--modify_execution_info` applies to the mnemonic level.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
native.apple_static_library(
    name = "lib",
    deps = [":swift_library_example"],
    tags = ["no-remote"],
)
```

The above won't apply the `no-remote` execution requirement to `lib`.

### What operating system are you running Bazel on?

macOS 11.6.1

### What's the output of `bazel info release`?

4.1.0

###  Have you found anything relevant by searching the web?

- https://github.com/bazelbuild/bazel/issues/8830

### Any other information, logs, or outputs that you want to share?

Seems that these native rules need to use `TargetUtils.getExecutionInfo(ruleContext.getRule(), ruleContext.isAllowTagsPropagation())`, like the built in CC and Java rules.

**Comment by brentleyjones** [comment]
Probably fixed in 5.0, which I couldn’t test on yet: https://github.com/bazelbuild/bazel/commit/31b689ba7238abe5629a0de87fb6e245910526bb. I’ll dig in more to ensure all of the actions I care about are covered now.

**Comment by brentleyjones** [comment]
This works in 5.0. Sorry for the noise!
